packages/block-editor/src/components/block-media-update-progress/index.native.js

```diff
@@ -0,0 +1,293 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import {
+	subscribeMediaUpload,
+	subscribeMediaSave,
+} from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+
+export const MEDIA_UPLOAD_STATE_UPLOADING = 1;
+export const MEDIA_UPLOAD_STATE_SUCCEEDED = 2;
+export const MEDIA_UPLOAD_STATE_FAILED = 3;
+export const MEDIA_UPLOAD_STATE_RESET = 4;
+
+export const MEDIA_SAVE_STATE_SAVING = 5;
+export const MEDIA_SAVE_STATE_SUCCEEDED = 6;
+export const MEDIA_SAVE_STATE_FAILED = 7;
+export const MEDIA_SAVE_STATE_RESET = 8;
+export const MEDIA_SAVE_FINAL_STATE_RESULT = 9;
+export const MEDIA_SAVE_MEDIAMODEL_CREATED = 10;
+
+export class BlockMediaUpdateProgress extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			progress: 0,
+			isSaveInProgress: false,
+			isSaveFailed: false,
+			isUploadInProgress: false,
+			isUploadFailed: false,
+		};
+
+		this.mediaUpload = this.mediaUpload.bind( this );
+		this.mediaSave = this.mediaSave.bind( this );
+	}
+
+	componentDidMount() {
+		this.addMediaUploadListener();
+		this.addMediaSaveListener();
+	}
+
+	componentWillUnmount() {
+		this.removeMediaUploadListener();
+		this.removeMediaSaveListener();
+	}
+
+	mediaIdContainedInMediaFiles( mediaId, mediaFiles ) {
+		if ( mediaId !== undefined && mediaFiles !== undefined ) {
+			for ( let i = 0; i < this.props.mediaFiles.length; i++ ) {
```

Comment:
Should this line be using the `mediaFiles` passed in as an argument instead of `this.props.mediaFiles`?

Reply:
Maybe this for-loop could be simplified using `Array.prototype.some()`

Reply:
nice ! also TIL - done in ca9392d53e